File: 377-combination-sum-iv/377-combination-sum-iv.py
class Solution:
    def combinationSum4(self, nums: List[int], target: int) -> int:
        
        dp = [0 for _ in range(target+1)]
        dp[0] = 1
        
        
        
        for j in range(1,target+1):
            for item in nums:
                if j-item < 0:continue
                dp[j] += dp[j-item]
        return dp[-1]
        

File: 377-combination-sum-iv/377-combination-sum-iv.java
class Solution {
    public int combinationSum4(int[] nums, int target) {
        int[] dp = new int[target + 1] ; 

        dp[0] = 1 ;
        int length = dp.length ; 
        for (int j = 0 ; j < dp.length ; j ++ ){
            for (int i : nums){
                if (j - i >= 0)
                    dp[j]  += dp[j-i];
            }
            // System.out.println(Arrays.toString(dp));
        }
        return dp[target];
    }
}